doc/internal/man3/OSSL_TIME.pod

```diff
@@ -0,0 +1,90 @@
+=pod
+
+=head1 NAME
+
+OSSL_TIME, OSSL_TIME_SECOND, OSSL_TIME_NEVER,
+ossl_time_now, ossl_time_time_to_timeval, ossl_time_compare,
+ossl_time_add, ossl_time_subtract
+- times and durations
+
+=head1 SYNOPSIS
+
+ #include "internal/time.h"
+
+ typedef OSSL_TIME;
+
+ #define OSSL_TIME_SECOND
+ #define OSSL_TIME_NEVER
+
+ OSSL_TIME ossl_time_now(void);
+ void ossl_time_time_to_timeval(OSSL_TIME t, struct timeval *out);
+
+ int ossl_time_compare(OSSL_TIME a, OSSL_TIME b);
+ OSSL_TIME ossl_time_add(OSSL_TIME a, OSSL_TIME b);
+ OSSL_TIME ossl_time_subtract(OSSL_TIME a, OSSL_TIME b);
+
+=head1 DESCRIPTION
+
+These functions allow the current time to be obtained and for basic
+arithmetic operations to be safely performed with times and durations.
+
+B<OSSL_TIME> is a type definition that holds a time.  It should not be
+assumed to be a structure, union or basic type.
+
+B<OSSL_TIME_SECOND> is an integer that indicates the precision of an
+B<OSSL_TIME>.  Specifically, it is the number of counts per second that
+a time can represent.
+
+B<OSSL_TIME_NEVER> is the largest representable B<OSSL_TIME>.  This value
+is returned when an overflow would otherwise occur.
+
+B<ossl_time_now> returns the current time.
+
+B<ossl_time_time_to_timeval> converts a time to a I<struct timeval>.
+
+B<ossl_time_compare> compares I<a> with I<b> and returns -1 if I<a>
+is smaller than I<b>, 0 if they are equal and +1 if I<a> is
+larger than I<b>.
+
+B<ossl_time_add> performs a saturating addition of the two times,
+returning I<a> + I<b>.
+If the summation would overflow B<OSSL_TIME_NEVER> is returned.
+
+B<ossl_time_subtract> performs a saturating subtraction of the two items,
+returning I<a> - I<b>.
+If the difference would be negative, the I<Epoch> (i.e. B<0>) is returned.
+
+=head1 NOTES
+
+The I<Epoch> is the start of the year 1970 UTC.  Times before this date
+are not supported.
+
+The largest representable time falls on the 22nd of July 2554.
```

Comment:
I don't think we should fix this as part of the defined API, or if we do we should choose a concrete representation of OSSL_TIME rather than being cagey about its representation, as this basically implies its representation. Rather we should specify a lower bound on the maximum representable value. I'd also express it as a duration first without mentioning any particular epoch ("This value can represent a duration of at least ~584 years.").

Reply:
Good point, changed.